nav2_navfn_planner/src/navfn_planner.cpp

```diff
@@ -134,6 +134,22 @@ nav_msgs::msg::Path NavfnPlanner::createPlan(
 #ifdef BENCHMARK_TESTING
   steady_clock::time_point a = steady_clock::now();
 #endif
+  unsigned int mx_start, my_start, mx_goal, my_goal;
+  if (!costmap_->worldToMap(start.pose.position.x, start.pose.position.y, mx_start, my_start)) {
+    throw nav2_core::StartOutsideMapBounds("Start outside map bounds");
+  }
+
+  if (!costmap_->worldToMap(goal.pose.position.x, goal.pose.position.y, mx_goal, my_goal)) {
+    throw nav2_core::GoalOutsideMapBounds("Goal outside map bounds");
+  }
+
+  if (costmap_->getCost(mx_start, my_goal) == nav2_costmap_2d::LETHAL_OBSTACLE) {
+    throw nav2_core::StartOccupied("Failed to create a unique pose path because of obstacles");
+  }
+
+  if (costmap_->getCost(mx_goal, my_goal) == nav2_costmap_2d::LETHAL_OBSTACLE) {
```

Comment:
Navfn will handle occupied goals actually, if tolerance is set to non-0. This should check if tolerance is set to 0, else this shouldn't be thrown